src/services/cache.rs

```diff
@@ -1,16 +1,16 @@
-use std::{thread, time::Duration, env, sync::Arc};
-use zmq::{Context, Socket, Message};
+use std::{thread, time::Duration, collections::HashMap};
 use crate::models::checkin::*;
-use threadpool::ThreadPool;
 use rusqlite::Connection;
 
+use super::apiclient::send_checkins;
+
 // Create a new database or connect to an existing one
 pub fn initialize_database() {
     // Tries to connect and if it can't connect then it will create a new database
     let db: Connection = Connection::open("./cache.sqlite").unwrap();
     let create_checkin_table_str = 
         "CREATE TABLE IF NOT EXISTS check_ins (
-            mac_address TEXT        NOT NULL,
+            id          TEXT        NOT NULL,
```

Comment:
I don't think you want to do this. Sqlite has a default field called ID anf this will overwrite it.